build-aux/m4/bitcoin_qt.m4

```diff
@@ -355,7 +355,6 @@ AC_DEFUN([_BITCOIN_QT_FIND_STATIC_PLUGINS],[
          PKG_CHECK_MODULES([QTFB], [Qt5FbSupport], [QT_LIBS="-lQt5FbSupport $QT_LIBS"])
                 fi
        if test "x$TARGET_OS" = xlinux; then
-         PKG_CHECK_MODULES([X11XCB], [x11-xcb], [QT_LIBS="$X11XCB_LIBS $QT_LIBS"])
```

Comment:
We can't just remove this, as it should still be necessary for non-depends builds. In other words, whatever we change this to needs to work before AND after removing the dependencies.

I think it should suffice to simply continue if the X11XCB check fails, and assume that qt was built without it as a dependency.

More correctly, we may be able to test a qt feature flag in one of the .pc files, or compile a stub that makes sure that QT_CONFIG(xcb_xlib) isn't set if libX11 is missing.

Reply:
> I think it should suffice to simply continue if the X11XCB check fails, and assume that qt was built without it as a dependency.

Played around with this a little bit...

Adding an empty last argument seems to trigger the default _action-if-not-found_ behaviour of "end the execution with an error for not having found the dependency" ([source](https://autotools.io/pkgconfig/pkg_check_modules.html))
```
PKG_CHECK_MODULES([X11XCB], [x11-xcb], [QT_LIBS="$X11XCB_LIBS $QT_LIBS"], [])
```

Adding a dummy last argument seems to work
```
PKG_CHECK_MODULES([X11XCB], [x11-xcb], [QT_LIBS="$X11XCB_LIBS $QT_LIBS"], [QT_LIBS="$QT_LIBS"])
```

I don't know autotools that well so perhaps @theuni can tell me what the idiomatic thing to do is here.

Reply:
I just looked into this. The .pc file for QTXCBQPA adds "-lX11-xcb" as necessary, so our pkg-config check is redundant anyway (unless it was done to fix an ordering problem, which doesn't seem to be the case).

So nevermind my complaint above, removal is fine.

Reply:
I double-checked @theuni's result with a Gitian build of 76e2cded477bc483ec610212bdadf21fe35292d4, which was before any XLib pickup failure. 

Here is `depends/i686-pc-linux-gnu/lib/pkgconfig/Qt5XcbQpa.pc`:
```
ubuntu@a661ab7477ba:~/build/bitcoin/depends/i686-pc-linux-gnu/lib/pkgconfig$ cat Qt5XcbQpa.pc
prefix=/home/ubuntu/build/bitcoin/depends/i686-pc-linux-gnu
exec_prefix=${prefix}
libdir=${prefix}/lib
includedir=${prefix}/include


Name: Qt5 XcbQpa
Description: Qt XcbQpa module
Version: 5.9.7
Libs: -L${libdir} -lQt5XcbQpa
Libs.private: -L/home/ubuntu/build/bitcoin/depends/i686-pc-linux-gnu/lib -lQt5ServiceSupport -lQt5ThemeSupport -lQt5DBus -lQt5EventDispatcherSupport -lQt5FontDatabaseSupport -L//home/ubuntu/build/bitcoin/depends/i686-pc-linux-gnu/lib -lfontconfig -lfreetype -lQt5Gui -lqtlibpng -lqtharfbuzz -lQt5Core -lm -lz -lqtpcre2 -lpthread -lX11-xcb -lX11 -lxcb-static -lxcb -ldl
Cflags: -I${includedir}/QtXcbQpa
Requires: Qt5Core Qt5Gui Qt5Core Qt5Gui Qt5ServiceSupport Qt5ThemeSupport Qt5EventDispatcherSupport Qt5FontDatabaseSupport
```

So, we should be good.